### PR TITLE
ES6-ify ListView Basics

### LINKED ISSUE #8184: Update ListView docs
If we're switching the docs to ES6 classes we need to update the `ListView` docs. They use `getInitialState` which doesn't work with ES6 classes.

### PR DESCRIPTION
**Pull request by JoelMarcey**
Fixes #8184

Test Plan:

http://localhost:8079/react-native/docs/basics-component-listview.html

screencapture-localhost-8079-react-native-docs-basics-component-listview-html-1466701988959

screenshot 2016-06-23 10 14 20

### CODE REVIEW

**Comment by facebook-github-bot-9** [comment]
By analyzing the blame information on this pull request, we identified @ericnakagawa and @lacker to be potential reviewers.

File: docs/Basics-Component-ListView.md [hunk 1]
```diff
@@ -20,20 +20,21 @@ This example creates a simple `ListView` of hardcoded data. It first initializes
 > A `rowHasChanged` function is required to use `ListView`. Here we just say a row has changed if the row we are on is not the same as the previous row.
 
 ```JavaScript
-import React from 'react';
+import React, {Component} from 'react';
```
Comment (artyomtrityak) [inline docs/Basics-Component-ListView.md]: just to be consistent can you please add spaces around Component? 
`import React, { Component } from 'react';`

**Comment by artyomtrityak** [comment]
Thanks! If it possible can you please do same for your another ES6-ify* reviews?

**Comment by JoelMarcey** [comment]
@artyomtrityak Done!

**Comment by artyomtrityak** [comment]
@JoelMarcey  thank you so much!

**Comment by caabernathy** [comment]
Looks great.

**Comment by caabernathy** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-5** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/260838494289090/int_phab) to review.

**Pull request closed without merging**